### PR TITLE
Fix plugin to work with passing podspecs

### PR DESCRIPTION
**Pull request by dnkoutso**
executing `pod dependencies Some.podspec` was not working since the `SourcesManager` code has changed in latest CocoaPods. 

Also performs a couple of other cleanup tasks.

### CODE REVIEW

File: lib/pod/command/dependencies.rb [hunk 1]
```diff
@@ -38,14 +38,8 @@ def validate!
         if @podspec_name
           require 'pathname'
           path = Pathname.new(@podspec_name)
-          if path.exist?
-            @podspec = Specification.from_file(path)
-          else
-            @podspec = SourcesManager.
```
Comment (dnkoutso) [inline lib/pod/command/dependencies.rb]: This was not working at all so I deleted it. It might be better to add in another PR.

So one can do `pod SomeRemote.podspec` and it gets pulled instead of being present locally.

File: lib/pod/command/dependencies.rb [hunk 2]
```diff
@@ -73,21 +67,23 @@ def dependencies
           end
 
           lockfile = Lockfile.generate(podfile, specs, {})
-          pods = lockfile.to_hash['PODS']
```
Comment (dnkoutso) [inline lib/pod/command/dependencies.rb]: unused assignment

File: lib/pod/command/dependencies.rb [hunk 3]
```diff
@@ -73,21 +67,23 @@ def dependencies
...
+          lockfile.to_hash['PODS']
         end
       end
 
       def podfile
         @podfile ||= begin
           if podspec = @podspec
             platform = podspec.available_platforms.first
-            platform_name, platform_version = platform.name, platform.deployment_target.to_s
```
Comment (dnkoutso) [inline lib/pod/command/dependencies.rb]: warning fix for parallel assignment

File: lib/pod/command/dependencies.rb [hunk 4]
```diff
@@ -73,21 +67,23 @@ def dependencies
...
-            sources = SourcesManager.all.map(&:url)
```
Comment (dnkoutso) [inline lib/pod/command/dependencies.rb]: this was not working anymore. now fixed.

File: lib/pod/command/dependencies.rb [hunk 5]
```diff
@@ -140,11 +136,6 @@ def podfile_dependencies
         Set.new(podfile.target_definitions.values.map { |t| t.dependencies.map { |d| d.name } }.flatten)
       end
 
-      # Returns a [String] of the names of dependencies specified in the podspec.
-      def podspec_dependencies
```
Comment (dnkoutso) [inline lib/pod/command/dependencies.rb]: unused method.

**Comment by segiddins** [comment]
Mind if I push a few changes? Mostly looks 💯

**Comment by dnkoutso** [comment]
Be my guest!

**Comment by segiddins** [comment]
@dnkoutso take a look, this should prevent a regression in functionality

**Comment by dnkoutso** [comment]
💯  I can try it locally real quick to ensure I can do `pod dependencies remote.podspec`

File: Gemfile [hunk 6]
```diff
@@ -1,4 +1,5 @@
 source 'https://rubygems.org'
 
-# Specify your gem's dependencies in cocoapods-dependencies.gemspec
 gemspec
+
+gem 'cocoapods', '~> 1.0'
```
Comment (dnkoutso) [inline Gemfile]: why wouldnt we want this to be in the gemspec?

**Comment by dnkoutso** [comment]
p.s I dont have access to merge here so I leave it up to you :)

**Comment by dnkoutso** [comment]
@segiddins should `pod dependencies CocoaLumberjack.podspec` work out of the box? It does not as I am seeing:
```
[!] Cannot find `CocoaLumberjack.podspec`.
```

**Comment by segiddins** [comment]
are you in a directory that contains a `CocoaLumberjack.podspec`?

**Comment by segiddins** [comment]
otherwise it'd de `pod dependencies CocoaLumberjack`

**Comment by dnkoutso** [comment]
Nope. I was under the impression that the `else` block in:
```ruby
          if path.file?
            @podspec = Specification.from_file(path)
          else
            sets = Config.
              instance.
              sources_manager.
              search(Dependency.new(@podspec_name))
            spec = sets && sets.specification
            @podspec = spec && spec.subspec_by_name(@podspec_name)
            raise Informative, "Cannot find `#{@podspec_name}`." unless @podspec
          end
```
Would go and fetch the podspec from the sources

**Comment by dnkoutso** [comment]
Ah yes! That was it! works.

**Comment by dnkoutso** [comment]
merge away.

**Comment by segiddins** [comment]
Will ty and merge & release tomorrow

**Comment by segiddins** [comment]
Released as 1.1.0
